tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/legalize_to_linalg.cc

```diff
@@ -1368,7 +1368,7 @@ DotOperationType GetDotOperationType(mhlo::DotOp dot_op) {
       shape_matches(lhs_shape[1], rhs_shape[0])) {
     return DotOperationType::kMatrixVector;
   }
-  if (rhs_shape.size() == 2 && rhs_shape.size() == 2 &&
+  if (lhs_shape.size() == 2 && rhs_shape.size() == 2 &&
```

Comment:
Can this show up in a test?

Reply:
hlo-legalize-to-linalg.mlir already has some test show up. [bug](https://github.com/google/iree/issues/8046) discussion,
I think when ```lhs_shape.size() == 1 && `rhs_shape.size() == 2 ``` will cause crash. But according to the [documentation](https://www.tensorflow.org/xla/operation_semantics#dot) of xla this doesn't happen, so I need to verify the dot op manually.and there is an obvious typo.

Reply:
I mean: can you include a test in this patch that would fail without the change but pass with the change?

Reply:
the dot wiil cause assert error.
```
%364 = "mhlo.dot"(%363, %arg4) : (tensor<87xf32>, tensor<87x32xf32>) -> tensor<32xf32>
```
But this is not compliant with xla semantics, xla expects a single dimension to be on the right, I'm not sure if mhlo strictly conforms to xla semantics, so I can't add this test,  Should I add dotop's verifier?

Reply:
Looking at the results, I think it should be the matrix dot vector, but their positions are reversed. I am following the iree community, how this ir is generated, because we lack the verifier, so this error occurs.